src/lib.rs

```diff
@@ -329,6 +329,16 @@ impl Config {
     /// otherwise cargo will link against the specified library.
     ///
     /// The given library name must not contain the `lib` prefix.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// gcc::Config::new()
+    ///             .file("src/foo.c")
+    ///             .shared_flag(true)
+    ///             .cpp_set_stdlib(Some("stdc++"))
+    ///             .compile("libfoo.so");
```

Comment:
Hm I'm realizing now these have `shared_flag`, but how come? I think this as-is will fail at runtime because this doesn't support creating `libfoo.so` afaik.